floor_generator/lib/writer/database_writer.dart

```diff
@@ -94,55 +94,44 @@ class DatabaseWriter implements Writer {
       ..name = 'migrations'
       ..type = refer('List<Migration>'));
 
-    final onConfigureParameter = Parameter((builder) => builder
-      ..name = 'onConfigure'
-      ..named = true
-      ..type = refer('sqflite.OnDatabaseConfigureFn'));
-
-    final onCreateParameter = Parameter((builder) => builder
-      ..name = 'onCreate'
-      ..named = true
-      ..type = refer('sqflite.OnDatabaseCreateFn'));
-
-    final onUpgradeParameter = Parameter((builder) => builder
-      ..name = 'onUpgrade'
-      ..named = true
-      ..type = refer('sqflite.OnDatabaseVersionChangeFn'));
+    final callbackParameter = Parameter((builder) => builder
+      ..name = 'callback'
+      ..type = refer('Callback'));
 
     return Method((builder) => builder
       ..name = 'open'
       ..returns = refer('Future<sqflite.Database>')
       ..modifier = MethodModifier.async
       ..requiredParameters.addAll([nameParameter, migrationsParameter])
-      ..optionalParameters.addAll([
-        onConfigureParameter,
-        onCreateParameter,
-        onUpgradeParameter,
-      ])
+      ..optionalParameters.addAll([callbackParameter])
       ..body = Code('''
           final path = join(await sqflite.getDatabasesPath(), name);
 
           return sqflite.openDatabase(
             path,
             version: ${database.version},
             onConfigure: (database) async {
-              if (onConfigure != null) {
-                await onConfigure(database);
-              }
               await database.execute('PRAGMA foreign_keys = ON');
             },
+            onOpen: (database) async {
+              if (callback?.onOpen != null) {
+                await callback.onOpen(database);
+              }
```

Comment:
This could be simplified with the following. Same applies for the other callbacks.

```dart
await callback?.onOpen?.call(database);
```

Do you have an idea how to only use `await` in case `onOpen` returns a `Future`?

Reply:
Maybe something like this would work, untested:

```dart
final result = callback?.onOpen?.call(database);
if (result is Future) {
  await result;
}
```

Reply:
But I believe having an `await` has no side effects if the callback function is not `async`.